pkg/ottl/ottlfuncs/func_parse_key_value_test.go

```diff
@@ -0,0 +1,270 @@
+package ottlfuncs
+
+import (
+	"context"
+	"testing"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+)
+
+func Test_parseKeyValue(t *testing.T) {
+	tests := []struct {
+		name           string
+		target         ottl.StringGetter[any]
+		delimiter      ottl.Optional[string]
+		pair_delimiter ottl.Optional[string]
+		expected       map[string]any
+	}{
+		{
+			name: "simple",
+			target: ottl.StandardStringGetter[any]{
+				Getter: func(ctx context.Context, tCtx any) (any, error) {
+					return "name=ottl func=key_value", nil
+				},
+			},
+			delimiter:      ottl.Optional[string]{},
+			pair_delimiter: ottl.Optional[string]{},
+			expected: map[string]any{
+				"name": "ottl",
+				"func": "key_value",
+			},
+		},
+		{
+			name: "large",
+			target: ottl.StandardStringGetter[any]{
+				Getter: func(ctx context.Context, tCtx any) (any, error) {
+					return `name=ottl age=1 job="software engineering" location="grand rapids michigan" src="10.3.3.76" dst=172.217.0.10 protocol=udp sport=57112 port=443 translated_src_ip=96.63.176.3 translated_port=57112`, nil
+				},
+			},
+			delimiter:      ottl.Optional[string]{},
+			pair_delimiter: ottl.Optional[string]{},
+			expected: map[string]any{
+				"age":               "1",
+				"port":              "443",
+				"dst":               "172.217.0.10",
+				"job":               "software engineering",
+				"location":          "grand rapids michigan",
+				"name":              "ottl",
+				"protocol":          "udp",
+				"sport":             "57112",
+				"src":               "10.3.3.76",
+				"translated_port":   "57112",
+				"translated_src_ip": "96.63.176.3",
+			},
+		},
+		{
+			name: "double quotes",
+			target: ottl.StandardStringGetter[any]{
+				Getter: func(ctx context.Context, tCtx any) (any, error) {
+					return `requestClientApplication="Mozilla/5.0 (Windows NT 6.1; WOW64; rv:40.0) Gecko/20100101 Firefox/40.0"`, nil
```

Comment:
Is there a test case with the pair_delimiter present in a quoted value? I didn't see one but I may have missed it.

Reply:
Yep, the test `quoted value contains delimiter and pair delimiter` covers that case. Although if you think it'd be better to split those 2 cases up/each case has their own specific test case I can certainly do so

Reply:
Nope, that's fine. I was just blind 🙈